cartography/intel/aws/ec2.py

```diff
@@ -847,6 +888,50 @@ def load_ec2_vpcs(neo4j_session, data, region, current_aws_account_id, aws_updat
         )
 
 
+TAG_RESOURCE_TYPE_MAPPINGS = {
+    'instance': {'label': 'EC2Instance', 'property': 'instanceid'},
+    'network-interface': {'label': 'NetworkInterface', 'property': 'id'},
+    'security-group': {'label': 'EC2SecurityGroup', 'property': 'id'},
+    'subnet': {'label': 'EC2Subnet', 'property': 'subnetid'},
+    'vpc': {'label': 'AWSVPC', 'property': 'id'},
+}
+
+TAG_INGEST_TEMPLATE = Template("""
+MATCH (n:$label{$property: {id}})
+WITH n
+MERGE (tag:Tag{key: {Key}})
+ON CREATE SET tag.firstseen = timestamp()
+WITH tag, n
+MERGE (n)<-[r:TAGGED]-(tag)
+ON CREATE SET r.firstseen = timestamp()
+SET r.value = {Value}, r.lastupdated = {aws_update_tag}
```

Comment:
Can you instead make `value` a property of the `Tag` node, itself?
```
MERGE (tag:{key: {Key}, value:{Value}})
```
That way, it might be simpler to do something like this
```
MATCH (tag:Tag)--[:TAGGED]->(n)
WHERE tag.key = 'Environment' and tag.value= 'Dev'
RETURN tag, n
```
, or even this
```
MATCH (tag:TAG)
WHERE tag.key = 'Environment'
RETURN tag
```

Reply:
Agree on making `value` a property of the node instead of the relationship.  @SecPrez knows more on this but there's something about properties on relationships being slow.